AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -253,10 +259,10 @@ - (void)setAsyncDataSource:(id<ASCollectionViewDataSource>)asyncDataSource
   // the (common) case of nilling the asyncDataSource in the ViewController's dealloc. In this case our _asyncDataSource
   // will return as nil (ARC magic) even though the _proxyDataSource still exists. It's really important to nil out
   // super.dataSource in this case because calls to _ASTableViewProxy will start failing and cause crashes.
+  _asyncDataSource = nil;
```

Comment:
Is this change actually necessary, given that it will be replaced / overwritten below anyway?  If it is necessary, could you add a comment that explains why?